graphbolt/src/index_select.cc

```diff
@@ -20,5 +22,33 @@ torch::Tensor IndexSelect(torch::Tensor input, torch::Tensor index) {
   return input.index({index.to(torch::kLong)});
 }
 
+std::tuple<torch::Tensor, torch::Tensor> IndexSelectCSC(
+    torch::Tensor indptr, torch::Tensor indices, torch::Tensor nodes) {
+  TORCH_CHECK(
+      indices.sizes().size() == 1, "IndexSelectCSC only supports 1d tensors");
+  if (indices.is_pinned() && utils::is_accessible_from_gpu(indptr) &&
+      utils::is_accessible_from_gpu(nodes)) {
+    GRAPHBOLT_DISPATCH_CUDA_ONLY_DEVICE(
+        c10::DeviceType::CUDA, "UVAIndexSelectCSC",
+        { return UVAIndexSelectCSCImpl(indptr, indices, nodes); });
+  } else if (
+      indices.device().type() == c10::DeviceType::CUDA &&
+      utils::is_accessible_from_gpu(indptr) &&
+      utils::is_accessible_from_gpu(nodes)) {
+    GRAPHBOLT_DISPATCH_CUDA_ONLY_DEVICE(
+        c10::DeviceType::CUDA, "nodesSelectCSC",
+        { return IndexSelectCSCImpl(indptr, indices, nodes); });
+  }
+  // @todo: The CPU supports only integer dtypes for indices tensor.
```

Comment:
If this is for CPU support, let's just raise an runtime_error("not implemented") here and follow up with another PR.

Reply:
I think the dispatch inside InSubgraph automatically raises an error like that.

Reply:
But yes, if we write not implemented, it is more clear.

Reply:
nit: Sorry for the unclear comment, I meant to remove the whole section that use FusedCSCSamplingGraph for CPU and discuss it in the separate PR.

Reply:
#6737 opened an issue.